site/source/docs/api_reference/wasm_workers.rst

```diff
@@ -342,6 +342,42 @@ table.
 
   </table>
 
+Wasm Workers stack size considerations
+======================================
+
+When instantiating a Wasm Worker, one has to create a memory array for the LLVM data stack for the created Worker. This data stack will generally consist only of local variables that have been "spilled" by LLVM into memory, e.g. to contain large arrays, structs, or other variables that are referenced by a memory address. This stack will not contain program flow information.
+
+Since WebAssembly does not support virtual memory, the size of the LLVM data stack that is defined both for Wasm Workers but also the main thread will not be possible to grow at runtime. So if the Worker (or the main thread) runs out of stack space, the program behavior will be undefined. Use the Emscripten linker flag -sSTACK_OVERFLOW_CHECK=2 to emit runtime stack overflow checks into the program code to detect these situations during development.
+
+Note that a stack overflow in this LLVM data stack does not imply stack overflow security vulnerabilities in the WebAssembly VM. The Wasm VM program flow stack is separate from this LLVM data stack, and guarded from access from Wasm code. However, if your threat model consists of executing untrusted program code, or executing a program from untrusted input inside the WebAssembly module, you may need to consider your security model to include safekeeping the LLVM data stack as well.
```

Comment:
Both `WebAssembly VM` and `Wasm VM` are used here.  Maybe just use one of them for consistency?